plugins/inputs/sqlserver/sqlserverqueries.go

```diff
@@ -1150,3 +1150,91 @@ FROM (
 	ORDER BY [record_id] DESC
 ) AS z
 `
+
+const sqlServerAvailabilityReplicaStates string = `
+IF SERVERPROPERTY('EngineEdition') NOT IN (2,3,4) BEGIN /*NOT IN Standard,Enterpris,Express*/
+	DECLARE @ErrorMessage AS nvarchar(500) = 'Telegraf - Connection string Server:'+ @@ServerName + ',Database:' + DB_NAME() +' is not a SQL Server Standard,Enterprise or Express. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END
+
+SELECT
+	'sqlserver_hadr_avreplica_states' AS [measurement],
+	REPLACE(@@SERVERNAME,'\',':') AS [sql_instance],
+	convert(nvarchar(36), hars.replica_id) as replica_id,
+	ar.replica_server_name,
+	convert(nvarchar(36), hars.group_id) as group_id,
+	ag.name AS group_name,
+	ar.replica_metadata_id,
+	availability_mode,
+	availability_mode_desc,
+	failover_mode,
+	failover_mode_desc,
+	session_timeout,
+	primary_role_allow_connections,
+	primary_role_allow_connections_desc,
+	secondary_role_allow_connections,
+	secondary_role_allow_connections_desc,
+	seeding_mode,
+	seeding_mode_desc,
+	basic_features,
+	is_distributed,
+	is_local,
+	role,
+	role_desc,
+	operational_state,
+	operational_state_desc,
+	connected_state,
+	connected_state_desc,
+	recovery_health,
+	recovery_health_desc,
+	synchronization_health,
+	synchronization_health_desc,
+	last_connect_error_number,
+	last_connect_error_description,
+	last_connect_error_timestamp
+from sys.dm_hadr_availability_replica_states AS hars
+inner join sys.availability_replicas AS ar on hars.replica_id = ar.replica_id
+inner join sys.availability_groups AS ag on ar.group_id = ag.group_id
+`
+
+const sqlServerDatabaseReplicaStates string = `
+IF SERVERPROPERTY('EngineEdition') NOT IN (2,3,4) BEGIN /*NOT IN Standard,Enterpris,Express*/
```

Comment:
Can you also add this check? only if availbility groups enabled
IF SERVERPROPERTY('EngineEdition') NOT IN (2,3,4)  or SERVERPROPERTY('IsHadrEnabled') = 0

Reply:
Fixed!

Reply:
reverted this change and did the IsHadrEnabled check before executing the query